astropy/cosmology/_io/mapping.py

```diff
@@ -153,7 +164,9 @@ def _rename_map(map, /, renames):
     return {renames.get(k, k): v for k, v in map.items()}  # dict separate from input
 
 
-def _get_cosmology_class(cosmology, params, /):
+def _get_cosmology_class(
+    cosmology: type[_CosmoT] | str | None, params: dict[str, Any], /
```

Comment:
Should this be
```suggestion
    cosmology: type[_CosmoT] | str | None, params: MutableMapping[str, Any], /
```

Reply:
As a private function we can type it specifically to how it's used. Currently it only ever receives a dict.

Reply:
Because this is a private function then we indeed don't have to be very forgiving with the types.